src/classifyInputString.cpp

```diff
@@ -0,0 +1,81 @@
+#include "classifyInputString.h"
+
+namespace {
+
+    InputParam classifyInputStringBase(const std::string & str) {
+
+        if(str.empty())
+            return unknown_param;
+
+        // if exactly 64 chars in length, it is likely a transaction id
+        if(str.length() == 64)
+            return txid_param;
+
+        // if it starts with certain chars, and is of a certain length, it may be a bitcoin address
+        if(str[0] == '1' || str[0] == '3' || str[0] == 'm' || str[0] == 'n' || str[0] == '2')
+            if(str.length() >= 26 && str.length() < 36)
+                return address_param;
+
+        // before testing for various txrefs, get rid of any unknown characters
+        std::string s = bech32::stripUnknownChars(str);
+
+        using namespace txref::limits;
+
+        // may be be txref (with leading HRP)
+        if(s.length() == TXREF_STRING_MIN_LENGTH ||
+           s.length() == TXREF_STRING_MIN_LENGTH_TESTNET)
+            return txref_param;
```

Comment:
At this point, we don't yet know the result of rejectBStringThatIsntWellFormed().  In considering BIP readability and our desired usage, does the order matter?  If order does not matter, then our lead-in to the BIP change might state: "among bech32 strings that are well formed, the following two length checks discriminate between references to transactions and references to transaction outputs..."

Reply:
I think I'm not sure what you mean here. Let's discuss tomorrow.